client/modules/User/components/SignupForm.jsx

```diff
@@ -43,16 +35,12 @@ function SignupForm() {
   return (
     <Form
       fields={['username', 'email', 'password', 'confirmPassword']}
-      validate={validateSignup}
+      validate={(values) => SignupFormInput.safeParse(values)}
```

Comment:
The `validate` function here needs to return an object of errors, where the keys are the fields and the values are the error messages.

Reply:
Ah ok this format is working:
```
username: z
    .string({
      required_error: i18n.t('ReduxFormUtils.errorEmptyUsername')
    })
    .max(20, { message: i18n.t('ReduxFormUtils.errorLongUsername') })
    .regex(/^[a-zA-Z0-9._-]{1,20}$/, {
      message: i18n.t('ReduxFormUtils.errorValidUsername')
    }),
  email: z
    .string({
      required_error: i18n.t('ReduxFormUtils.errorEmptyEmail')
    })
    .email({ message: i18n.t('ReduxFormUtils.errorInvalidEmail') }),
```

Putting an error on the `min(1)` didn't do it.  Seems like we need to use `required_error` to handle empty field errors.

When playing around it is helpful to set your language to something other than English so you can see that the messages are getting translated.